### PR TITLE
Fix/timeservice max time

### LINKED ISSUE #248: TimeService max time is never set
Currently the max time in the TimeService is hard-coded to 10,000. This cannot be intended, though no TODO is to be found 🤔. Should probably be set once the timing file has been parsed.

### PR DESCRIPTION
**Pull request by cwognum**
#### What have I done
First and foremost, I would like to come clean and admit to not mentioning the issue number in all my commit messages. I hope you (I'm talking to you, almighty @Dacaspex) are so kind to forgive me. I shall mention the issue number in this PR to make up for my unlimited stupidity. 

Fixes #248 (sets the maximum time upon reading a timing file and changes the slider to be able to work with a maximum time of 0)

#### Testing
Run all test cases. Once we have proper input files, it ofcourse makes sense to test using those. 

#### Addresses SRD section(s)
Not sure, but there probably is something in there about the maximum time and the slider. I looked for it, but could not find it that quickly. @Astrejoe, do you know what needs to be updated?

### CODE REVIEW

**Comment by nolmoonen** [comment]
SRD section would concern "Animation Toolbar" I think.

File: src/app/backend/time/time.service.spec.ts [hunk 1]
```diff
@@ -233,4 +238,10 @@ describe('TimeService', () => {
         service.setPlaybackSpeed(-25);
         expect(service.playbackValid()).toEqual(true);
     });
+
+    it('Should give a percentage done of zero when maximum time equals zero', () => {
```
Comment (nolmoonen) [inline src/app/backend/time/time.service.spec.ts]: capital letter

File: src/app/backend/time/time.service.spec.ts [hunk 2]
```diff
@@ -212,9 +217,9 @@ describe('TimeService', () => {
     });
 
     it('should give the correct place percentage of the animation that is done.', () => {
```
Comment (nolmoonen) [inline src/app/backend/time/time.service.spec.ts]: dot

File: src/app/backend/time/time.service.spec.ts [hunk 3]
```diff
@@ -124,6 +127,7 @@ describe('TimeService', () => {
     });
 
     it('CurrentTime can not go below 0', () => {
```
Comment (nolmoonen) [inline src/app/backend/time/time.service.spec.ts]: capital letter

File: src/app/backend/dispatcher/dispatcher.service.spec.ts [hunk 4]
```diff
@@ -49,6 +49,7 @@ describe('DispatcherService', () => {
 
     it('#update should progress time',
```
Comment (nolmoonen) [inline src/app/backend/dispatcher/dispatcher.service.spec.ts]: Please remove unused import.

File: src/app/backend/file-parser/timing-parser.service.ts [hunk 5]
```diff
@@ -5,13 +5,14 @@ import { TimingEntry } from '../data-representation/timing-entry';
 import { SheetTiming } from '../data-representation/sheet-timing';
```
Comment (nolmoonen) [inline src/app/backend/file-parser/timing-parser.service.ts]: Please remove unused import.

File: src/app/backend/file-parser/timing-parser.service.ts [hunk 6]
```diff
@@ -48,6 +49,8 @@ export class TimingParserService {
 
         const timingId = this.sheetContainer.getNumTimings();
 
+        let maxTime = -1;
```
Comment (nolmoonen) [inline src/app/backend/file-parser/timing-parser.service.ts]: Why is this -1, what is the meaning of the value?

File: src/app/backend/time/time.service.spec.ts (hunk 1 above)
Comment (Extremelyd1) [reply src/app/backend/time/time.service.spec.ts]: Removed the cApiTaL.

File: src/app/backend/time/time.service.spec.ts (hunk 2 above)
Comment (Extremelyd1) [reply src/app/backend/time/time.service.spec.ts]: Done.....

File: src/app/backend/time/time.service.spec.ts (hunk 3 above)
Comment (Extremelyd1) [reply src/app/backend/time/time.service.spec.ts]: Rewritten the sentence.

File: src/app/backend/dispatcher/dispatcher.service.spec.ts (hunk 4 above)
Comment (Extremelyd1) [reply src/app/backend/dispatcher/dispatcher.service.spec.ts]: There is no unused import.

File: src/app/backend/file-parser/timing-parser.service.ts (hunk 5 above)
Comment (Extremelyd1) [reply src/app/backend/file-parser/timing-parser.service.ts]: Also no unused import.

File: src/app/backend/file-parser/timing-parser.service.ts (hunk 6 above)
Comment (Extremelyd1) [reply src/app/backend/file-parser/timing-parser.service.ts]: Changed initial value to `undefined` and rewritten the checks.

**Pull request merged**